bundles/mapping/mapmodule/plugin/wmslayer/WmsLayerPlugin.ol.js

```diff
@@ -153,16 +174,85 @@ Oskari.clazz.define(
                 }
                 // Set min max zoom levels that layer should be visible in
                 zoomLevelHelper.setOLZoomLimits(layerImpl, _layer.getMinScale(), _layer.getMaxScale());
+                // Adjust min zoom level based on timeseries metadata toggle level
+                const metadata = this._getTimeSeriesMetadata(layer);
+                if (metadata.toggleLevel > -1) {
+                    layerImpl.setMinZoom(metadata.toggleLevel);
+                }
                 this.mapModule.addLayer(layerImpl, !keepLayerOnTop);
                 // gather references to layers
                 olLayers.push(layerImpl);
 
+                // The timeseries metadata layer is a different type of layer and
+                // must be managed separately, thus it cannot be added to olLayers
+                // array, in which the layers are handled as WMS layers.
+                this._timeSeriesMetadataLayer = this._createTimeSeriesMetadataLayer(metadata);
+                if (this._timeSeriesMetadataLayer) {
+                    this.mapModule.addLayer(this._timeSeriesMetadataLayer, !keepLayerOnTop);
+                }
+
                 this._log.debug('#!#! CREATED ol/layer/TileLayer for ' + _layer.getId());
             }
             // store reference to layers
             this.setOLMapLayers(layer.getId(), olLayers);
         },
 
+        _getTimeSeriesMetadata: function (layer) {
+            const options = layer.getOptions();
+            const timeseries = options.timeseries || {};
+            return timeseries.metadata || {};
+        },
+
+        _createTimeSeriesMetadataLayer: function (metadata) {
+            const { layer: layerId, toggleLevel } = metadata;
+            if (!layerId || !toggleLevel || toggleLevel < 0) {
+                return;
+            }
+
+            const source = new olVectorSource({
+                format: new olGeoJSON(),
+                url: (extent) =>
+                    Oskari.urls.getRoute('GetWFSFeatures', {
+                        id: layerId,
+                        bbox: extent.join(','),
+                        srs: Oskari.getSandbox().getMap().getSrsName()
+                    }),
+                strategy: olBboxStrategy
+            });
+
+            const style = this._getMetadataLayerStyle(layerId);
+            const layer = new olVectorLayer({
+                source,
+                style,
+                maxZoom: toggleLevel
+            });
+            return layer;
+        },
+
+        _getMetadataLayerStyle: function (layerId) {
+            const service = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
+            const layer = service.findMapLayer(layerId);
+            const styles = layer.getOptions().styles;
```

Comment:
Could use:
const layerStyleDef = layer.getStyleDef(this._timeSeriesMetadataLayerStyleName) || {};
const featureStyleDef = layerStyleDef.featureStyle || {};

With custom styles:
"timeseriesStyle": {
  "featureStyle" : {
     "stroke":...

https://oskari.org/documentation/examples/oskari-style#vector-layer-styling